src/tools/pkcs11-tool.c

```diff
@@ -2577,6 +2602,24 @@ static void decrypt_data(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 		mech.pParameter = iv;
 		mech.ulParameterLen = iv_size;
 		break;
+	case CKM_AES_GCM:
+		use_aead_buffer = TRUE;
+		iv_size = 16;
+		iv = get_iv(opt_iv, &iv_size);
+		gcm_params.pIv = iv;
+		gcm_params.ulIvLen = iv_size;
+		gcm_params.ulIvBits = iv_size * 8; // no one seems to know what this is for
```

Comment:
The section 2.13.5 AES GCM and CCM Mechanism parameters says:

> `ulIvBits`        length of initialization vector in bits. Do no use ulIvBits to specify the length of the initialization vector, but ulIvLen instead.

so I would drop this.

https://docs.oasis-open.org/pkcs11/pkcs11-curr/v3.0/os/pkcs11-curr-v3.0-os.html#_Toc30061261